patches/tModLoader/Terraria/ModLoader/ProjectileLoader.cs

```diff
@@ -295,18 +295,24 @@ private class HookList
 			}
 		}
 
-		private static HookList HookCanDamage = AddHook<Func<Projectile, bool>>(g => g.CanDamage);
+		private static HookList HookCanDamage = AddHook<Func<Projectile, bool?>>(g => g.CanDamage);
+
+		public static bool? CanDamage(Projectile projectile) {
+			bool? result = null;
 
-		public static bool CanDamage(Projectile projectile) {
-			if (projectile.modProjectile != null && !projectile.modProjectile.CanDamage()) {
-				return false;
-			}
 			foreach (GlobalProjectile g in HookCanDamage.arr) {
-				if (!g.Instance(projectile).CanDamage(projectile)) {
-					return false;
+				bool? canDamage = g.Instance(projectile).CanDamage(projectile);
+
+				if (canDamage.HasValue) {
+					if (!canDamage.Value) {
+						return false;
+					}
+
+					result &= canDamage.Value;
 				}
```

Comment:
This doesn't work, because the starting value is null. This should just be = true

Reply:
what even is &= as an operator :thinkies:

Reply:
a = a & b.